src/test/java/com/ge/predix/uaa/token/lib/HttpServletRequestUtilTest.java

```diff
@@ -107,6 +107,15 @@ public void testGetSubdomainFromOneHeader(final String requestHeader, final List
         Assert.assertEquals(subdomain, subdomainValue);
     }
 
+    @Test(dataProvider = "domainDataProvider")
+    public void testGetSubdomainFromDomain(final String serverName, final List<String> serviceConfigBaseDomains,
```

Comment:
please add a new test to verify if 2 different zoneIds are specified in subdomain and header, then the one in subdomain is returned.